astropy/units/physical.py

```diff
@@ -386,47 +398,55 @@ def _dimensionally_compatible_unit(obj):
         else:
             return NotImplemented
 
-    def _dimensional_analysis(self, other, operation):
+    def _dimensional_analysis(
+        self, other: object, operation: str
+    ) -> PhysicalType | NotImplementedType:
         other_unit = self._dimensionally_compatible_unit(other)
         if other_unit is NotImplemented:
             return NotImplemented
         other_unit = _replace_temperatures_with_kelvin(other_unit)
         new_unit = getattr(self._unit, operation)(other_unit)
         return new_unit.physical_type
 
-    def __mul__(self, other):
+    def __mul__(
+        self, other: PhysicalType | core.UnitBase | numbers.Real | str
+    ) -> PhysicalType:
         return self._dimensional_analysis(other, "__mul__")
 
-    def __rmul__(self, other):
+    def __rmul__(self, other: PhysicalType | core.UnitBase | str) -> PhysicalType:
```

Comment:
The type of the `other` parameter in `__mul__` includes `numbers.Real`, but in `__rmul__` it does not. This is deliberate because Mypy would complain otherwise. My superficial understanding is that the reason is in how `numbers.Real` is annotated. `numbers` does not work very well with type checking anyways.